src/test/kotlin/org/wycliffeassociates/otter/common/data/workbook/TestAssociatedAudio.kt

```diff
@@ -0,0 +1,64 @@
+package org.wycliffeassociates.otter.common.data.workbook
+
+import com.jakewharton.rxrelay2.ReplayRelay
+import org.junit.Assert
+import org.junit.Test
+import org.wycliffeassociates.otter.common.assertEqualsForEach
+import org.wycliffeassociates.otter.common.data.model.MimeType
+import java.io.File
+import java.time.LocalDate
+
+class TestAssociatedAudio {
+    private fun createTakeWithNum(num: Int): Take =
+        Take(
+            "name",
+            File("."),
+            num,
+            MimeType.WAV,
+            LocalDate.now()
+        )
+
+    private fun createTakesRelay(numTakes: Int): ReplayRelay<Take> {
+        val takesRelay = ReplayRelay.create<Take>()
+        for (i in 1..numTakes) {
+            takesRelay.accept(createTakeWithNum(i))
+        }
+        return takesRelay
+    }
+
+    @Test
+    fun testInsertTake() {
+        val audio = AssociatedAudio(createTakesRelay(0))
+
+        val take1 = createTakeWithNum(1)
+        audio.insertTake(take1)
+        Assert.assertEquals(1, audio.takes.values.size)
+        Assert.assertTrue(audio.takes.contains(take1).blockingGet())
+
+        val take2 = createTakeWithNum(2)
+        audio.insertTake(take2)
+        Assert.assertEquals(2, audio.takes.values.size)
+        Assert.assertTrue(audio.takes.contains(take2).blockingGet())
+    }
+
+    @Test
+    fun testGetAllTakes() {
+        val numTakes = 10
+        val audio = AssociatedAudio(createTakesRelay(numTakes))
+        val array = audio.getAllTakes()
+        Assert.assertEquals(numTakes, array.size)
+    }
+
+    @Test
+    fun testGetNewTakeNumber() {
+        val inputs = listOf(0, 1, 2, 9, 10, 99, 100)
+        val outputs = inputs.map { it + 1 }
+        inputs
+            .zip(outputs)
+            .toMap()
+            .assertEqualsForEach {
+                val audio = AssociatedAudio(createTakesRelay(it))
+                audio.getNewTakeNumber().blockingGet()
+            }
+    }
```

Comment:
The way this is implemented, deleted takes' take numbers are still treated as "used" take numbers. So, if we have 3 takes, and delete take 3, the next take we create would be take 4. In reality, maybe this should be the new take 3. But there are a few problems...
- First, we could be overwriting the old take 3 file if we use the same filename. We might not want that. (?) It depends on what the purpose of keeping around deleted takes is.
- Second, the filenames in the database (Take_entity) table would be the same for the old take 3 and the new take 3

Now, the take numbers that the _view_ displays to the user doesn't necessarily have to correspond to the take numbers in the filenames. But this can be confusing if they are ever aware of the filename of the take they are listening to. When they open, say, ocenaudio, to edit a take, you can see the take filename at the top.

Another more complicated case that I have been thinking about: If we have 5 takes and delete take 2, would we have to shift every take number down?